### ISSUE TITLE
RRD4J will stop working with Java 9

### ISSUE THREAD

**Issue by huxi** [rrd4j/rrd4j#72]
org.rrd4j.core.RrdNioBackend is using internal API that will be removed in Java 9.

```
$ jdeps -jdkinternals rrd4j-2.2.jar
rrd4j-2.2.jar -> /Library/Java/JavaVirtualMachines/jdk1.8.0_45.jdk/Contents/Home/jre/lib/rt.jar
   org.rrd4j.core.RrdNioBackend (rrd4j-2.2.jar)
      -> sun.misc.Cleaner                                   JDK internal API (rt.jar)
      -> sun.nio.ch.DirectBuffer                            JDK internal API (rt.jar)

Warning: JDK internal APIs are unsupported and private to JDK implementation that are
subject to be removed or changed incompatibly and could break your application.
Please modify your code to eliminate dependency on any JDK internal APIs.
For the most recent update on JDK internal API replacements, please check:
https://wiki.openjdk.java.net/display/JDK8/Java+Dependency+Analysis+Tool

JDK Internal API                         Suggested Replacement
----------------                         ---------------------
sun.misc.Cleaner                         Use java.lang.ref.PhantomReference @since 1.2
```

**Comment by analytically** [comment]
Still unresolved: http://bugs.java.com/view_bug.do?bug_id=4724038

**Comment by huxi** [comment]
Ah, OK. So that's the reason. I didn't understand the issue from looking at the code.

It would still be possible to do this via reflection to prevent a `ClassNotFoundException` if `sun.nio.ch.DirectBuffer` is missing.

**Comment by evernat** [comment]
Yes, [JDK-4724038](https://bugs.openjdk.java.net/browse/JDK-4724038) is not yet resolved, but [JDK-8171377](https://bugs.openjdk.java.net/browse/JDK-8171377) is resolved and available in JDK9 build 150.
It seems that sun.misc.Unsafe.theUnsafe.invokeCleaner(buffer) and reflection is the way to go for JDK 9.
Otherwise it will not work using JDK 9.

**Comment by fbacchella** [comment]
I have a test code in 604209f01552, in branch https://github.com/rrd4j/rrd4j/tree/ForJava9.

**Comment by tuxedo0801** [comment]
pls. see my comment on commit https://github.com/rrd4j/rrd4j/commit/604209f01552fe5a1d24d0245257301851f0d027

**Comment by fbacchella** [comment]
Should be solved by https://github.com/rrd4j/rrd4j/commit/4b791a7ac03c26ff159f97fa0716c0d28aa15a45.

**Comment by fbacchella** [comment]
I will issue a version 3.3 when java 11 is out.

**Comment by bplump** [comment]
Hi @fbacchella - Now that [Java 11 is out](https://www.infoq.com/news/2018/09/java11-released), could you please release version 3.3 when you have some time? Thank you and awesome to see this was fixed before the release!

**Comment by fbacchella** [comment]
Done in 799bd39
